chain/client/src/sync/adapter.rs

```diff
@@ -31,86 +29,108 @@ pub enum SyncMessage {
     SyncDone(SyncShardInfo),
 }
 
-struct ActorHandler {
-    /// Address of actor mailbox
-    addr: actix::Addr<SyncActor>,
-    /// Thread handler of actor
-    arbiter: actix::Arbiter,
+pub struct SyncActorHandler {
+    pub client_sender: Sender<SyncMessage>,
+    pub network_sender: Sender<StateSyncResponse>,
+    pub shutdown: Mutex<Box<dyn FnMut() + Send>>,
+}
+
+impl Drop for SyncActorHandler {
+    fn drop(&mut self) {
+        std::mem::replace(self.shutdown.lock().unwrap().deref_mut(), Box::new(|| {}))();
+    }
 }
 
 /// Manager for state sync threads.
 /// Offers functions to interact with the sync actors, such as start, stop and send messages.
 pub struct SyncAdapter {
     /// Address of the sync actors indexed by the SharUid
-    actor_handler_map: HashMap<ShardUId, ActorHandler>,
+    actor_handler_map: HashMap<ShardUId, SyncActorHandler>,
     /// Message channel for client
     client_adapter: Sender<SyncMessage>,
     /// Message channel with network
     network_adapter: Sender<PeerManagerMessageRequest>,
+    /// Function to make an actor
+    actor_maker: Arc<
+        dyn Fn(ShardUId, Sender<SyncMessage>, Sender<PeerManagerMessageRequest>) -> SyncActorHandler
```

Comment:
nit: `Fn(ShardUId, Sender<SyncMessage>, Sender<PeerManagerMessageRequest>) -> SyncActorHandler  + Send + Sync` is used in multiple places, so maybe let's use type alias for that